azure/scope/identity.go

```diff
@@ -212,7 +212,7 @@ func (p *AzureCredentialsProvider) GetTenantID() string {
 	return p.Identity.Spec.TenantID
 }
 
-func createAzureIdentityWithBindings(ctx context.Context, azureIdentity *infrav1.AzureClusterIdentity, clusterMeta metav1.ObjectMeta,
+func createAzureIdentityWithBindings(ctx context.Context, azureIdentity *infrav1.AzureClusterIdentity, resourceManagerEndpoint, activeDirectoryEndpoint string, clusterMeta metav1.ObjectMeta,
```

Comment:
is it worth refactoring this slightly to add a test?  This function gets the structs as well as creates objects. If it only returned the structs a test could be written the logic to do creation would be hoisted up where other dependencies are.

Reply:
another option would be to use the fake kubeClient